microsetta_admin/server.py

```diff
@@ -658,6 +658,114 @@ def metadata_pulldown():
                                search_error=errors)
 
 
+@app.route('/submit_daklapack_order', methods=['GET'])
+def submit_daklapack_order():
+    error_msg_key = "error_message"
+
+    def return_error(msg):
+        return render_template('submit_daklapack_order.html',
+                               **build_login_variables(),
+                               error_message=msg)
+
+    status, dak_articles_output = APIRequest.get(
+        '/api/admin/daklapack_articles')
+    if status >= 400:
+        return return_error("Unable to load daklapack articles list.")
+
+    status, projects_output = _get_projects(include_stats=False,
+                                            is_active=True)
+    if status >= 400:
+        return return_error(projects_output[error_msg_key])
+
+    return render_template('submit_daklapack_order.html',
+                           **build_login_variables(),
+                           error_message=None,
+                           dummy_status=DUMMY_SELECT_TEXT,
+                           dak_articles=dak_articles_output,
+                           contact_phone_number=SERVER_CONFIG[
+                               "order_contact_phone"],
+                           projects=projects_output['projects'])
+
+
+@app.route('/submit_daklapack_order', methods=['POST'])
+def post_submit_daklapack_order():
+    def return_error(msg):
+        return render_template('submit_daklapack_order.html',
+                               **build_login_variables(),
+                               error_message=msg)
+
+    error_message = success_message = headers = None
+    expected_headers = ["firstName", "lastName", "address1", "insertion",
+                        "address2", "postalCode", "city", "state",
+                        "country", "countryCode"]
+
+    # get required fields; cast where expected by api
+    phone_number = request.form['contact_phone_number']
+    project_ids_list = list(map(int, request.form.getlist('projects')))
+    dak_article_code = int(request.form['dak_article_code'])
+    file = request.files['addresses_file']
+
+    # get optional fields or defaults
+    description = request.form.get('description')
+    fedex_ref_1 = request.form.get('fedex_ref_1')
+    fedex_ref_2 = request.form.get('fedex_ref_2')
+    fedex_ref_3 = request.form.get('fedex_ref_3')
+    fulfillment_hold_msg = request.form.get('fulfillment_hold_msg')
+
+    try:
+        addresses_df = pd.read_excel(file)
```

Comment:
`xlrd` cannot read `.xlsx` and `openpyxl` cannot read `.xls`. :/ I think the following should cover the typical uses here.

```suggestion
    try:
        if file.endswith('xls'):
            addresses_df = pd.read_excel(file)
        else:
            addresses_df = pd.read_excel(file, engine='openpyxl')
```

Reply:
Done :)